### ISSUE TITLE
text length error

### ISSUE THREAD

**Issue by Nipi64310** [UKPLab/sentence-transformers#104]
```
model = SentenceTransformer(modules=[word_embedding_model, pooling_model,dense])
tri_reader = TripletReader('dataset/xxx')
train_data = SentencesDataset(tri_reader.get_examples('triplets5.tsv'), model=model)
train_dataloader = DataLoader(train_data, shuffle=False, batch_size=batch_size)
```
use triplet loss
I debug the params find all text in a batch becomes the same length 6，like the first sentence in first pairs length？:
```train_data[0]``` is
([[2769, 4638, 1355, 6573, 749, 720],
  [711, 784, 720, 2769, 4638, 6820, 3766, 1355, 6573, 1435],
  [6716,7770,  122,   127,   129,   860,   7028,   122,   122,   121,   2682,   6206,   2160,   3351,
   3126,   3362]],
 tensor(1))

```
for i in train_dataloader:
    break
print(i)
```
[[[tensor([2769, 2769, 2769, 2769, 2769, 2769, 2769, 1355, 1355, 1567,  671,  671,
           6435,  872, 2571, 2582]),
   tensor([4638, 4638, 4638, 4692, 4692, 4692, 4692, 1139, 1139, 4381, 4684, 4684,
           7309, 6929, 6853,  720]),
   tensor([1355, 1355, 1355, 4289, 4289, 4289, 4289, 3341, 3341, 2692, 3227, 3227,
           2769, 6804, 2582, 6820]),
   tensor([6573, 6573, 6573, 3837, 3837, 3837, 3837,  749,  749, 8043, 4850, 4850,
            743, 1355,  720, 3766]),
   tensor([ 749,  749,  749, 3227, 3227, 3227, 3227, 1408, 1408, 2347, 3005, 3005,
           4638, 6573,  671, 3119]),
   tensor([ 720,  720,  720, 4850, 4850, 4850, 4850, 8043, 8043, 5307, 3119, 3119,
            691,  749, 4684, 1168])],
  [tensor([ 711,  122, 2769, 2347, 2347, 6820, 1377, 5543, 2644, 2347, 3221, 5439,
           6435, 6929,  671, 6821]),
   tensor([ 784,  122, 6573, 5307, 5307,  809,  809,  679, 1962, 5307,  679, 3352,
           2226, 3193,  702, 4658]),
   tensor([ 720, 1384, 3219,  743, 2571,  711, 2226, 5543, 8024,  673, 3221,  872,
           2571, 4157, 4851, 2582]),
   tensor([2769,  743, 1921,  749,  671, 3219, 2571, 2376, 2644, 1921, 1355, 6820,
           2376, 1355, 2876,  720]),
   tensor([4638, 4638, 2347,  697, 1453, 1921, 1355, 2769, 2157,  749, 6573, 3766,
           2769, 6573,  749, 6820])],
  [tensor([6716,  872,  872,  166,  800, 6821, 5050, 4289, 1525, 3918, 2769, 1920,
           6929, 2798, 6370, 2190]),
   tensor([7770, 2157,  812,  154, 3683,  816,  749, 1501, 7027, 5905, 2218, 2207,
           6821, 4959, 1296, 4638]),
   tensor([ 122, 1567, 3221, 1962, 6772, 6137, 8024, 6574, 1355, 5682, 3221, 1962,
            702,  671, 5356, 8024]),
   tensor([ 127, 3198, 3343,  719, 4607, 6136, 5018, 7030, 6573,  833, 4509, 5272,
           5587, 1921, 1384, 1166]),
   tensor([ 129,  952, 2336, 3300,  852, 4638,  671,  924, 1557,  679, 6435, 5310,
           1741, 2218, 3221, 3018])]],
 tensor([1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1])]

```
train_dataloader = DataLoader(train_data, shuffle=False, batch_size=batch_size,collate_fn=model.smart_batching_collate)
the result is correct , 
but in examples, the code have not  collate_fn=model.smart_batching_collate
```

and another bugs,  use cnn encoder, kernel size only can all even or all odds , like [1,3,5] or [2,4,6],
if use [2,3,4,5] , will get error
